### PR TITLE
Support using label names when changing issue labels

### LINKED ISSUE #30917: Failure in Gitea API when using labeler Github action
### Description

I am using **[actions/labeler@v5](https://github.com/actions/labeler)**  with Gitea's Actions, the runner is running **gitea/act_runner:nightly-dind-rootless**, the labels are not placed on the issue & the actual workflow step is failing with the following message, added below 

It's coming from the Gitea API, & i'd like to understand what how can i help to improve this.

Error:

```
The configuration file (path: .gitea/labeler.yaml) was found locally, reading from the file
::error::HttpError: []: json: cannot unmarshal number " into Go struct field IssueLabelsOption.Labels of type int64
::error::[]: json: cannot unmarshal number " into Go struct field IssueLabelsOption.Labels of type int64
```

the job looks very simple:

```
jobs:
  labeler:
    name: Labeler
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Labeler
        uses: actions/labeler@v5
        with:
          configuration-path: .gitea/labeler.yaml
          sync-labels: true
```

labeler.yaml

```
Documentation:
- changed-files:
  - any-glob-to-any-file: docs/**

```      

EDIT
from gitea's log:

```sh
...eb/routing/logger.go:102:func1() [I] router: completed PUT /api/v1/repos/mariusrugan/example-repo/issues/104/labels for 1.2.3.4:37056, 422 Unprocessable Entity in 20.9ms @ v1/api.go:689(v1.Routes.func1.10.1.1.2.2.bind[...])
```

Many thanks in advance!

### Gitea Version

1.21.11

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Docker container

### Database

SQLite

### PR DESCRIPTION
**Pull request by Zettat123**
Resolve #30917

Make the APIs for adding labels and replacing labels support both label IDs and label names so the [`actions/labeler`](https://github.com/actions/labeler) action can work in Gitea.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: modules/structs/issue_label.go [hunk 1]
```diff
@@ -47,8 +47,9 @@ type EditLabelOption struct {
 
 // IssueLabelsOption a collection of labels
 type IssueLabelsOption struct {
-	// list of label IDs
-	Labels []int64 `json:"labels"`
+	// Labels can be a list of intergers representing label IDs
```
Comment (kemzeb) [inline modules/structs/issue_label.go]: Small typo nitpick:
```suggestion
	// Labels can be a list of integers representing label IDs
```

Looks like you would also need to regenerate this in the swagger .tmpl file

File: routers/api/v1/repo/issue_label.go [hunk 2]
```diff
@@ -317,7 +319,32 @@ func prepareForReplaceOrAdd(ctx *context.APIContext, form api.IssueLabelsOption)
...
 
-	labels, err := issues_model.GetLabelsByIDs(ctx, form.Labels, "id", "repo_id", "org_id", "name", "exclusive")
+	var (
+		labelIDs   []int64
+		labelNames []string
+	)
+	for _, label := range form.Labels {
+		rv := reflect.ValueOf(label)
+		switch rv.Kind() {
+		case reflect.Float64:
```
Comment (kemzeb) [inline routers/api/v1/repo/issue_label.go:329]: LGTM! Seeing `reflect.Float64` be used here and having the "label IDs" test pass puzzled me for a bit but after reading some of the `jsoniter` code and `encoding/json` [docs](https://pkg.go.dev/encoding/json#Unmarshal) this is expected behavior when dealing with `any` and JSON numbers.

**Review by kemzeb — APPROVED** [review]

File: routers/api/v1/repo/issue_label.go [hunk 3]
```diff
@@ -317,7 +319,32 @@ func prepareForReplaceOrAdd(ctx *context.APIContext, form api.IssueLabelsOption)
...
+			labelIDs = append(labelIDs, int64(rv.Float()))
+		case reflect.String:
+			labelNames = append(labelNames, rv.String())
```
Comment (wolfogre) [inline routers/api/v1/repo/issue_label.go:332]: Maybe we should handle the default case for strange input like `[[1]]`.

**Review by wolfogre — APPROVED** [review]
LGTM.

**Pull request merged**